docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc

```diff
@@ -1,3 +1,52 @@
+= Transition from Java EE to Jakarta EE
+
+We have migrated our codebase from Java EE (Enterprise Edition) to Jakarta EE, which brings various changes into Keycloak.
+Java EE was a widely adopted technology stack for developing enterprise applications in Java.
+However, due to certain legal and licensing issues, Java EE was transferred to the Eclipse Foundation and renamed as Jakarta EE.
+
+We have upgraded all Jakarta EE specifications in order to support Jakarta EE 10, such as:
+
+* Jakarta Persistence 3.1
+* Jakarta RESTful Web Services 3.1
+* Jakarta Mail API 2.1
+* Jakarta Servlet 6.0
+* Jakarta Activation 2.1
+* Other
+
+Jakarta EE 10 provides a modernized, simplified, lightweight approach to building cloud-native Java applications.
+The main changes provided within this initiative are changing the namespace from `javax.\*` to `jakarta.*`.
+It does not apply for `javax.*` packages provided directly in the JDK, such as `javax.security`, `javax.net`, `javax.crypto`, etc.
+
+You can be affected by these changes in your custom extensions, providers or JPA entities.
+
+== Deprecated Adapters stay on Java EE
+
+As most of our OIDC and SAML client adapters are already deprecated, we will not upgrade them to support Jakarta EE.
+However, client adapters with Java EE support can still communicate with the Keycloak server.
+
+=  We have upgraded to Quarkus 3!
+
+We have upgraded the Quarkus Java framework, which lives underneath Keycloak and brings many benefits.
+Quarkus 3 continues the tradition of propelling Java development by moving fast and providing a cutting-edge user experience with the latest technologies.
+It continues to improve overall performance and efficiency.
+
+Quarkus 3 is based on Jakarta EE 10, the same as Keycloak, creating smooth interoperability between them.
+In addition, it contains Eclipse MicroProfile 6, which aligns with Jakarta EE 10 Core Profile.
+The central part of the Quarkus 3 upgrade is built-in support for JPA 3.1 and Hibernate ORM 6.
+
+=== Stop using `quarkus.hibernate-orm.*` properties
+For Quarkus 3, it is impossible to simultaneously specify Hibernate ORM properties for datasources via the `persistence.xml` file and Quarkus properties.
+As for Keycloak, these properties are specified via the `persistence.xml` file, and it is required to specify the additional properties there.
```

Comment:
I believe, we still plan to address this for 22: https://github.com/keycloak/keycloak/issues/19427
@ahus1 Can you please confirm?

Reply:
I'd keep the information here as it reflects the current state. If the issue is solved for KC 22, we can remove it from the migration guide. @vmuzikar @ahus1 WDYT?

Reply:
@mabartos +1 - if we choose to update it and find the time to address it, the docs will change. For now, I'd rather have the docs in main match the state we have in main - and this is what @mabartos prepared here.